#!/usr/bin/env python
PACKAGE = "iiwa_impedance_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("stiffness", double_t, 0, "Joint stiffness", 100, 0, 500)
gen.add("damping_ratio", double_t, 0, "Joint damping ratio", 0.7, 0, 1)

gen.add("seperate_joints", bool_t, 0, "Mode where stiffness for seperate joints can be defined",  False)
gen.add("stiffness_joint_1", double_t, 0, "Joint stiffness in joint 1", 0, 0, 500)
gen.add("stiffness_joint_2", double_t, 0, "Joint stiffness in joint 2", 0, 0, 500)
gen.add("stiffness_joint_3", double_t, 0, "Joint stiffness in joint 3", 0, 0, 500)
gen.add("stiffness_joint_4", double_t, 0, "Joint stiffness in joint 4", 0, 0, 500)
gen.add("stiffness_joint_5", double_t, 0, "Joint stiffness in joint 5", 0, 0, 500)
gen.add("stiffness_joint_6", double_t, 0, "Joint stiffness in joint 6", 0, 0, 500)
gen.add("stiffness_joint_7", double_t, 0, "Joint stiffness in joint 7", 0, 0, 500)
gen.add("damping_ratio_joint_1", double_t, 0, "Joint damping ratio", 0.7, 0, 1)
gen.add("damping_ratio_joint_2", double_t, 0, "Joint damping ratio", 0.7, 0, 1)
gen.add("damping_ratio_joint_3", double_t, 0, "Joint damping ratio", 0.7, 0, 1)
gen.add("damping_ratio_joint_4", double_t, 0, "Joint damping ratio", 0.7, 0, 1)
gen.add("damping_ratio_joint_5", double_t, 0, "Joint damping ratio", 0.7, 0, 1)
gen.add("damping_ratio_joint_6", double_t, 0, "Joint damping ratio", 0.7, 0, 1)
gen.add("damping_ratio_joint_7", double_t, 0, "Joint damping ratio", 0.7, 0, 1)

gen.add("set_new_reference_pose", bool_t, 0, "Set current joint pose to be the new reference joint pose",  False)
gen.add("set_new_home_pose", bool_t, 0, "Set current joint pose to be the new reference joint home pose",  False)
gen.add("set_reference_to_home_pose", bool_t, 0, "Set current joint reference pose to be the new reference end joint home pose",  False)

gen.add("q_d_joint_1", double_t, 0, "desired configuration for joint 1", 0, -2.96706, 2.96706)
gen.add("q_d_joint_2", double_t, 0, "desired configuration for joint 2", 0, -2.0944, 2.0944)
gen.add("q_d_joint_3", double_t, 0, "desired configuration for joint 3", 0, -2.96706, 2.96706)
gen.add("q_d_joint_4", double_t, 0, "desired configuration for joint 4", 0, -2.0944, 2.0944)
gen.add("q_d_joint_5", double_t, 0, "desired configuration for joint 5", 0, -2.96706, 2.96706)
gen.add("q_d_joint_6", double_t, 0, "desired configuration for joint 6", 0, -2.0944, 2.0944)
gen.add("q_d_joint_7", double_t, 0, "desired configuration for joint 7", 0, -3.0543262, 3.0543262)


exit(gen.generate(PACKAGE, "iiwa_impedance_control", "JointImpedanceController"))
